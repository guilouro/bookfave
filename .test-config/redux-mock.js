import React from 'react'
import { createStore, combineReducers, applyMiddleware } from 'redux'
import { render } from 'react-testing-library'
import { Provider } from 'react-redux'

export const mockStore = mock => {
  const reducers = {}

  Object.keys(mock).forEach(key => {
    reducers[key] = () => mock[key]
  })

  return createStore(combineReducers(reducers))
}

export const renderWithRedux = (component, store) =>
  render(<Provider store={store}>{component}</Provider>)
