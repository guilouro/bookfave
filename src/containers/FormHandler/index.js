import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import Forms from '../../components/Forms'
import { addFavorite, filterByTag } from '../../actions/favorites'

const FormHandler = ({ onAddFavorite, onFilterByTag, filterTag }) => {
  const handleAdd = ({ data }) => {
    onAddFavorite({
      title: data.title,
      link: data.link,
      tags: data.tags
    })
  }

  const handleSearch = tagName => {
    onFilterByTag(tagName.toLowerCase())
  }

  return (
    <Forms onAdd={handleAdd} onSearch={handleSearch} filterTag={filterTag} />
  )
}

FormHandler.propTypes = {
  onAddFavorite: PropTypes.func.isRequired,
  onFilterByTag: PropTypes.func.isRequired,
  filterTag: PropTypes.string
}

FormHandler.defaultProps = {
  filterTag: ''
}

const mapStateToProps = ({ favorites }) => ({
  filterBy: favorites.filterTag
})

export default connect(
  mapStateToProps,
  {
    onAddFavorite: addFavorite,
    onFilterByTag: filterByTag
  }
)(FormHandler)
