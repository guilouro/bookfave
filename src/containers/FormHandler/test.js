import React from 'react'
import FormHandler from '.'
import { mockStore, renderWithRedux } from '../../../.test-config/redux-mock'

const store = mockStore({
  favorites: {
    filterTag: '',
    entities: {},
    ids: []
  }
})

describe('<FormHandler />', () => {
  it('Should render correctly', () => {
    const { container } = renderWithRedux(<FormHandler />, store)
    expect(container.firstChild).toMatchSnapshot()
  })
})
