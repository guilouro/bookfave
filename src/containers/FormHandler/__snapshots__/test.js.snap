// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormHandler /> Should render correctly 1`] = `
<div
  class="sc-bdVaJa gtODql"
>
  <div
    class="sc-bwzfXH dOEENz"
  >
    <button
      class="sc-htpNat sc-bxivhb fSgGqO"
    >
      <svg
        height="20px"
        viewBox="0 0 26 26"
        width="20px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.583 20.083c-4.687 0-8.5-3.812-8.5-8.5 0-4.687 3.813-8.5 8.5-8.5 4.688 0 8.5 3.813 8.5 8.5 0 4.688-3.812 8.5-8.5 8.5m13.752 3.249l-4.81-4.811a11.265 11.265 0 0 0 2.391-6.938C22.916 5.334 17.832.25 11.583.25 5.334.25.25 5.334.25 11.583c0 6.25 5.084 11.334 11.333 11.334 2.615 0 5.02-.9 6.937-2.392l4.811 4.81a1.413 1.413 0 0 0 2.004 0 1.415 1.415 0 0 0 0-2.003"
          fill="currentColor"
        />
      </svg>
    </button>
    <button
      class="sc-htpNat sc-bxivhb fzDMrC"
    >
      <svg
        height="20px"
        viewBox="0 0 26 26"
        width="20px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.121 11.72h-3.84V7.879a1.28 1.28 0 0 0-2.561 0v3.84H7.879a1.28 1.28 0 0 0 0 2.561h3.84v3.841a1.28 1.28 0 0 0 2.561 0v-3.84h3.841a1.28 1.28 0 0 0 0-2.561zm5.121 7.682a3.845 3.845 0 0 1-3.84 3.84H6.598a3.845 3.845 0 0 1-3.84-3.84V6.598a3.845 3.845 0 0 1 3.84-3.84h12.804a3.845 3.845 0 0 1 3.84 3.84v12.804zM19.402.197H6.598A6.409 6.409 0 0 0 .197 6.598v12.804a6.409 6.409 0 0 0 6.401 6.401h12.804a6.409 6.409 0 0 0 6.401-6.401V6.598A6.409 6.409 0 0 0 19.402.197z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <form
    novalidate=""
  >
    <div
      class="sc-ifAKCX iPZVBd"
    >
      <input
        class="sc-EHOje gmOzGi"
        name="title"
        placeholder="Title"
        required=""
        value=""
      />
      <input
        class="sc-EHOje gmOzGi"
        name="link"
        placeholder="Link"
        required=""
        value=""
      />
      <input
        class="sc-EHOje gmOzGi"
        name="tags"
        placeholder="Tags"
        value=""
      />
      <button
        type="submit"
      >
        Add
      </button>
    </div>
  </form>
</div>
`;
